OSSM-1094 Use bcrypt for htpasswd, stop mounting rawpassword

cam-garrison opened the pull request:
See previous PR #1056, re-targeting for 2.4 release.

Will make necessary changes (run make gen to apply changes to resources/v2.4/) when v2.4 exists in the resources folder (likely near the end of the rebase cycle?)

from previous PR:

### Changes:

- Change to using bcrypt instead of SHA-1 for htpasswd.

- Update the oauth-proxy version from 4.4 -> 4.9 (first version to support bcrypt)

- Stop mounting the raw htpasswd in our grafana-proxy and prometheus-proxy containers.

### Tested:

- OpenShift upgrading from operator 2.2 to 2.3 with this change and it worked as expected (keeping SHA-1 but nothing breaks)
- Fresh install of this operator with SMCP 2.3 where it generates a bcrypt password, doesn't mount in prometheus-proxy and grafana-proxy as intended.

luksa committed:
OSSM-2190 Don't reconcile SMMR with invalid name

luksa committed:
OSSM-2232 Don't reconcile SMM with invalid name

luksa committed:
OSSM-2189 Reconcile only first SMCP when multiple SMCPs exist in namespace

cam-garrison committed:
use bcrypt to hash htpasswd

cam-garrison committed:
use bcrypt to hash htpasswd

cam-garrison committed:
update oauth-proxy version to 4.9

cam-garrison committed:
update oauth-proxy version to 4.9

cam-garrison committed:
only mount auth to proxy containers

cam-garrison committed:
only mount auth to proxy containers

cam-garrison committed:
apply gci, gufumpt

cam-garrison committed:
apply gci, gufumpt

cam-garrison committed:
reorder imports for linter

cam-garrison committed:
reorder imports for linter

bmangoen committed:
OSSM-2126 Configurable Grafana storage capacity

cam-garrison committed:
OSSM-2241: remove 2.0 as a control plane option (#1062)

* remove 2.0 as a control plane option in menu

* run make gen, updating 2.3 manifests

dgn committed:
OSSM-534 Support ingress.hosts for grafana and prometheus (#1052)

bmangoen committed:
Merge branch 'maistra-2.3' of https://github.com/maistra/istio-operator into OSSM-2126

bmangoen committed:
Using patch-charts.sh for changes

jwendell committed:
Bump to 2.3.1 (#1064)

dgn committed:
Pull in maistra-2.2 chart changes (#1066)

dgn committed:
OSSM-2187 Allow usage of spec.proxy.networking.protocol.autoDetect (#1049)

This removes the 'validation' that was effectively removing support for protocol sniffing and allows usage of the fields in 2.2 and 2.3 control planes. We previously removed this feature from OSSM 2.0 because of security concerns, however I cannot find any documentation of these concerns and upstream has been enabling it by default since Istio 1.6, so I consider it safe enough to allow usage. Note that we will still default to disabling it in order not to change behavior in any existing deployments.

bmangoen committed:
Merge pull request #1061 from bmangoen/OSSM-2126

cam-garrison committed:
OSSM-2221 remove ignorenamespace label from control plane ns [maistra-2.3] (#1060)

* remove ignorenamespace label from control plane ns

* remove ignore ns label from test

dgn reviewed (commented):
lgtm so far, but haven't actually tested it yet. The only thing that gives me a slight headache is the updaete of oauth-proxy to the 4.9 version- I wonder if that increases our minimum supported OCP version to 4.9? but even so, in 2.4 that should be fine

luksa committed:
OSSM-2188 Reconcile SMMR even when SMCP is missing (#1063)

* Extract getRequiredNamespaces()

* Reconcile SMMR even if SMCP doesn't exist

This ensures that the ServiceMeshMember resources created by the controller are deleted when the SMCP is deleted.

* Move SMM listing closer to where it's being used

luksa committed:
Fix linter issues

jwendell committed:
Remove vendor

luksa committed:
Add push target to Makefile (#1076)

jwendell committed:
Initial support for 2.4

cam-garrison committed:
rebase onto 2.4, make gen

cam-garrison committed:
Merge branch 'bcrypt-2.3' of https://github.com/cam-garrison/istio-operator into bcrypt-2.3

cam-garrison committed:
remove changes in v2.3 templates

cam-garrison commented:
closing and reopening for 2.4. rebasing was a poor idea here ✖_✖
